code/modules/client/preference/preferences.dm

```diff
@@ -595,9 +644,51 @@ GLOBAL_LIST_INIT(special_role_times, list( //minimum age (in days) for accounts
 				added_cost = 0
 			else
 				type_blacklist += G.main_typepath
-
 		if((total_cost + added_cost) > max_gear_slots)
 			continue // If the final cost is too high, don't add the item.
 		active_character.loadout_gear += G.type
 		total_cost += added_cost
 	return total_cost
+
+
+/datum/preferences/proc/init_keybindings(overrides, raw)
+	if(raw)
+		try
+			overrides = json_decode(raw)
+		catch
+			overrides = list()
+	keybindings = list()
+	keybindings_overrides = overrides
+	for(var/kb in GLOB.keybindings)
+		var/datum/keybinding/KB = kb
+		var/list/keys = (overrides && overrides[KB.name]) || KB.keys
+		for(var/key in keys)
+			LAZYADD(keybindings[key], kb)
+
+	parent?.update_active_keybindings()
```

Comment:
At this point, would it just be worth not doing this if the parent is null? When would that even happen?

Reply:
I'm not sure if/when prefs get deleted if a client disconnects, best have this safety in just incase?